src/dotnet/GeneratorAgent/src/Azure.Tools.GeneratorAgent/Agent/AgentResponseParser.cs

```diff
@@ -0,0 +1,193 @@
+using System.Text.Json;
+using Azure.Tools.GeneratorAgent.Models;
+using Microsoft.Extensions.Logging;
+using Azure.Tools.ErrorAnalyzers;
+
+namespace Azure.Tools.GeneratorAgent.Agent
+{
+    /// <summary>
+    /// Service for parsing and processing AI agent responses in plain text JSON format
+    /// </summary>
+    internal class AgentResponseParser
```

Comment:
What do you think of making this a static `AgentResponseParserHelpers` class and take out the ILogger stuff? Instead the caller can handle logging and this can just either return or throw an exception